tests/app_helper.hpp

```diff
@@ -41,15 +41,15 @@ class TempFile {
 };
 
 inline void put_env(std::string name, std::string value) {
-#ifdef _MSC_VER
+#ifdef _WIN32
```

Comment:
I am not very familiar with non-MSVC compilers on Windows anymore, so never know what is best here. I'll assume that non-MSVC compilers are still able to access the Windows calls, rather than GNU counterparts? It would be nice to add a tests to AppVeyor (or Azure) someday (not as part of this PR, though).

Reply:
The issue was MSYS, which is mingw,  deep in some of the libraries is uses some of the windows subsystems, so has most of the posix calls,  but when you get down into environmental variables you use the windows calls,  which I think was the case here.